### ISSUE TITLE
SPICE is no longer available in RHEL9/CentOS9 Stream

### ISSUE THREAD

**Issue by cfergeau** [dmacvicar/terraform-provider-libvirt#961]
### Linux distribution

``` Centos9 Stream/ RHEL 9 ```

### Provider and libvirt versions

```
terraform/providers/libvirt/go.mod:require github.com/dmacvicar/terraform-provider-libvirt v0.6.12
```

## Checklist

- [ ] Is it a bug or something that does not work as expected? Please make sure you fill the version information below:

## Description of Issue/Question

terraform-provider-libvirt uses SPICE  by default for its `libvirtxml.DomainGraphic`, but this has been deprecated/removed in RHEL9/CentOS9 Stream, see https://github.com/openshift/installer/pull/5622
```
[DEBUG] libvirt_domain.bootstrap: apply errored, but we're indicating that via the Error pointer rather than returning it: Error defining libvirt domain: virError(Code=67, Domain=10, Message='unsupported configuration: spice graphics are not supported with this QEMU')
```

terraform-libvirt-provider could parse the equivalent of `virsh domcapabilities` to make a better decision regarding which graphics element to add by default:
```
$ virsh domcapabilities
<domainCapabilities>
<!-- [...] -->
  <devices>
    <!-- [...] -->
    <graphics supported='yes'>
      <enum name='type'>
        <value>sdl</value>
        <value>vnc</value>
        <value>spice</value>
        <value>egl-headless</value>
      </enum>
    </graphics>
  </devices>
</domainCapabilities>
```
Or it could switch to VNC by default instead of SPICE.

**Comment by scabala** [comment]
I do not think this is actionable - default setting of one Linux distribution should not dictate defaults here. As a matter of fact, it's just a default that's easily changeable.
I believe this issue should be closed.

**Comment by dmacvicar** [comment]
ℹ️ ℹ️ ℹ️ ℹ️ ℹ️ ℹ️

This issue report is relevant to the legacy version of the provider, which is now in the v0.8 branch of this repository.

Future development is based on a new provider, which is not compatible with this one, nor does share code.

As the new provider solves many issues with the legacy, and to make development in my free time more enjoyable, I have decided to close all bugs for the legacy provider, and to ask to check if the bug would apply to the new one. This also to encourage trying the new version.

- https://registry.terraform.io/providers/dmacvicar/libvirt/latest

and check the documentation:

- https://github.com/dmacvicar/terraform-provider-libvirt/blob/main/README.md
- https://registry.terraform.io/providers/dmacvicar/libvirt/latest/docs

You are free to reopen the bug for v0.8. It may encourage someone to fix it. My efforts will go into the new provider.

I ask you to check the new provider and re-evaluate this bug. 🙏
